src/mds/SessionMap.h

```diff
@@ -291,6 +295,7 @@ class Session : public RefCountedObject {
 
   void touch_cap(Capability *cap) {
     session_cache_liveness.hit(1.0);
+    readdir_caps_throttle.hit(1.0);
```

Comment:
This needs to be hit in a separate method for readdir specifically. `touch_cap` will be called anytime the capability is manipulated (not just created).

Reply:
Look at the end of `Server::handle_client_readdir`. `numfiles` is what you'd want to hit the decay counter with.